### PR TITLE
add update_mesh method

### PR DESCRIPTION
**Pull request by kcpevey**
# Description

Adds a new method: `layers.vectors._update_mesh` which reduces duplicate code and makes the codebase cleaner. 

It calls the `generate_vector_meshes` method and sets the results into variables stored on the class.

This also will be beneficial for the `async_slicing` work being proposed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# References

# How has this been tested?

- [x] all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/layers/vectors/vectors.py
Comment (andy-sweet) [inline napari/layers/vectors/vectors.py]: Should this also be added to `_update_mesh`? It seems like it's mutated every time to keep the `_dims_displayed` last used to index `data`.

**Review by andy-sweet — CHANGES_REQUESTED** [review]
Looks good to me so far.

It looks like we could also use `_update_mesh` in `__init__` and in the `data` setter.

I guess some people might push back against using it in `__init__` because it's generally more clear to assign attributes directly there. But I would probably make an exception in this case.

Alternatively, we could encapsulate the mesh in a `dataclass` and then have a method that returns a new instance of that instead of mutating `_mesh_vertices`, `_mesh_triangles`, and maybe `_displayed_stored`.

File: napari/layers/vectors/vectors.py
Comment (kcpevey) [reply napari/layers/vectors/vectors.py]: yes, that's true. done.

**Comment by kcpevey** [comment]
I'd be happy to encapsulate into a `dataclass` if that's what people would prefer. I'll remove the draft in hopes of opening it up for more feedback.

**Review by andy-sweet — APPROVED** [review]
Looks good to me now. Would be good to get another core-dev to review and merge this - @brisvag maybe?

File: napari/layers/vectors/vectors.py
Comment (brisvag) [inline napari/layers/vectors/vectors.py]:
```suggestion
        self._data = data
        self._mesh_vertices = None
        self._mesh_triangles = None
        self._displayed_stored = None
```

**Review by brisvag — APPROVED** [review]
Looks great to me! One thing that makes me slightly squeamish is initializing private attributes in a method, so I suggested a change to be more explicit.

**Review by kevinyamauchi — APPROVED** [review]
Thanks, @kcpevey ! This looks good to me. I assume the tests will continue to pass after the latest change. Happy to merge once they finish!

**Comment by kevinyamauchi** [comment]
CI is passing and we have three approvals, so I am merging. Thanks again, @kcpevey !

**Pull request merged**
